components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/kmclient/model/BearerInterceptor.java

```diff
@@ -0,0 +1,23 @@
+package org.wso2.carbon.apimgt.impl.kmclient.model;
+
+import feign.RequestInterceptor;
+import feign.RequestTemplate;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.recommendationmgt.AccessTokenGenerator;
+
+public class BearerInterceptor implements RequestInterceptor {
+
+    private AccessTokenGenerator accessTokenGenerator;
+
+    public BearerInterceptor(AccessTokenGenerator accessTokenGenerator) {
+
+        this.accessTokenGenerator = accessTokenGenerator;
+    }
+
+    @Override
+    public void apply(RequestTemplate requestTemplate) {
+
+        String accessToken = accessTokenGenerator.getAccessToken(APIConstants.KEY_MANAGER_OAUTH2_REST_API_MGT_SCOPES);
+        requestTemplate.header("Authorization", "Bearer " + accessToken);
+    }
```

Comment:
we could use constants in the APIConstant class. AUTHORIZATION_HEADER_DEFAULT, AUTHORIZATION_BEARER are there